src/store/dialogsReducer.js

```diff
@@ -47,17 +47,24 @@ const initialState = {
 };
 
 const dialogsReducer = (state = initialState, action) => {
+  const stateCopy = {
+    ...state,
+    messages: [...state.messages],
+  };
+
   switch (action.type) {
-    case UPDATE_NEW_MESSAGE_BODY:
-      state.newMessageBody = action.body;
-      return state;
+    case UPDATE_NEW_MESSAGE_BODY: {
+      stateCopy.newMessageBody = action.body;
```

Comment:
This is a mutation.
Consider using spread operator.

Reply:
/fxd